backend/__tests__/posts.js

```diff
@@ -402,4 +402,124 @@ describe('Posts', () => {
+
+                expect(response.statusCode).toBe(404)
+            })
+        })
+
+        describe('when a valid user is sharing a post that exists', () => {
+            it('should return response code of 201', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const postCreationResponse = await request(server)
+                    .post('/api/posts')
+                    .set('Authorization', `Bearer ${authToken}`)
+                    .send({
+                        text_content: 'what is the meaning of life?',
+                        parent: null,
+                    })
+
+                const response = await request(server)
+                    .post(`/api/posts/${postCreationResponse.body.id}/share`)
+                    .set('Authorization', `Bearer ${authToken}`)
+
+                expect(response.statusCode).toBe(201)
+            })
+        })
+    })
+
+    describe('DELETE /posts/:id/share', () => {
+        describe('when not authenticated', () => {
+            it('should return response code of 400', async () => {
+                const response = await request(server).post(
+                    '/api/posts/1/share'
+                )
+                expect(response.statusCode).toBe(400)
+            })
+        })
+
+        describe('when the user is unsharing a post that they did not share', () => {
+            it('should return response code of 404', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const response = await request(server)
+                    .delete(`/api/posts/99/share`)
+                    .set('Authorization', `Bearer ${authToken}`)
+
+                expect(response.statusCode).toBe(404)
+            })
+        })
+
+        describe('when a valid user is unsharing a post', () => {
+            it('should return response code of 200', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                await request(server)
+                    .post(`/api/posts/2/share`)
+                    .set('Authorization', `Bearer ${authToken}`)
```

Comment:
Here you're sharing a post that you haven't created, wouldnt that cause an error if we don't have that post?

Also, would it be possible to only send requests to the endpoints that we're testing? The rest can be added directly into the db using models.create(), so that changes to the POST /share endpoint does not cause it to fail

Reply:
Okay I fixed it.